src/vanilla/utils/proxyMap.ts

```diff
@@ -88,10 +88,8 @@ export function proxyMap<K, V>(entries?: Iterable<[K, V]> | undefined | null) {
     has(key: K) {
       const map = getMapForThis(this)
       const exists = map.has(key)
-      if (!exists) {
-        // eslint-disable-next-line @typescript-eslint/no-unused-expressions
-        this.index // touch property for tracking
-      }
+      // eslint-disable-next-line @typescript-eslint/no-unused-expressions
+      this.data.length // touch property for tracking
       return exists
```

Comment:
Let's do this:
```suggestion
      return map.has(key)
```

Reply:
Yeah I was planning on it actually. Just thought I'd push this up first to get the patch out. I didn't have a whole lot of time today with other work. Do you want me to push up a different pr for the tests or do you want it in this one?

Reply:
I added the tests and pushed up